### PR TITLE
Chart: Default airflow version to 2.4.1

### PR DESCRIPTION
**Pull request by ephraimbuddy**

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Hmmm @ephraimbuddy - this failure is a but suspicious. I think (hope) it is just a way of retrieving the version in the test is not compatible wiht some recent "version" refactor @uranusjr ?

**Review by potiuk — APPROVED** [review]

File: chart/Chart.yaml [hunk 1]
```diff
@@ -46,23 +46,23 @@ annotations:
       url: https://airflow.apache.org/docs/helm-chart/1.6.0/
   artifacthub.io/screenshots: |
     - title: DAGs View
-      url: https://airflow.apache.org/docs/apache-airflow/2.3.4/_images/dags.png
+      url: https://airflow.apache.org/docs/apache-airflow/2.4.0/_images/dags.png
     - title: Tree View
```
Comment (jedcunningham) [inline chart/Chart.yaml]: We should add the datasets related screenshots too.

Comment (jedcunningham) [inline chart/Chart.yaml]:
```suggestion
    - title: Datasets View
      url: https://airflow.apache.org/docs/apache-airflow/2.4.0/_images/datasets.png
    - title: Tree View
```

Comment (jedcunningham) [inline chart/Chart.yaml]:
```suggestion
    - title: Grid View
```

**Review by jedcunningham — CHANGES_REQUESTED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by Aakcht** [comment]
Hello, some python unit tests here are probably failing because of [this PR](https://github.com/apache/airflow/pull/24496).  I added some TODOs there that should be changed after 2.4.0 update in the chart:

https://github.com/apache/airflow/pull/24496/files#diff-d4dbf2216fff973901e9385afa008421089e057ece11d15c1a9e54a0d391142fR38

https://github.com/apache/airflow/pull/24496/files#diff-e09db3d8e07ac0c27076914b006ad8e670430273378af3d9f3d9d6d83f2a1c6aR114

**Review by jedcunningham — APPROVED** [review]

File: tests/charts/test_basic_helm_chart.py [hunk 2]
```diff
@@ -37,8 +37,7 @@ def _get_values_with_version(self, values, version):
         return values
 
     def _get_object_count(self, version):
-        # TODO remove default from condition after airflow update
-        if version == "2.3.2" or version == "default":
+        if version == "2.3.2":
```
Comment (dstandish) [inline tests/charts/test_basic_helm_chart.py:40]: could do `semver.VersionInfo.parse(version) < (2, 4)` for better clarity

**Review by dstandish — APPROVED** [review]

File: tests/charts/test_basic_helm_chart.py (hunk 2 above)
Comment (jedcunningham) [reply tests/charts/test_basic_helm_chart.py:40]: After looking at this again with fresh eyes, I like `if version == "2.3.2"` better than `if version != "default" and semver.VersionInfo.parse(version) < (2, 4)`. Seems more straight forward.

**Pull request merged**